Makes benches work again after simplification

mortendahl opened the pull request:
Solves https://github.com/mortendahl/rust-paillier/issues/10

gbenattar commented:
Looks good to me, one concern with the following bench:
```
benchmark_group!(zk_2048,
    self::bench_zk_proof_challenge<KeySize2048>,
    self::bench_zk_proof_prove<KeySize2048>,
    self::bench_zk_proof_prove_and_verify<KeySize2048>,
    self::bench_zk_proof_prove_all<KeySize2048>
);
```

This will take "forever", can we limit the number of iteration done by the bench?

mortendahl commented:
hmm, yes, but we need to use a fork of bencher afaik -- checking..

mortendahl commented:
@gbenattar I've switched to the snips bencher, which will automatically lower the number of iterations if a bench is taking too long. it could be a bit more aggressive (without looking at the code it seems to only lower if taking more than 1s) but that's probably a future PR :)

also renamed the correct-key proof trait to match with the correct-opening trait
